flink-staging/flink-streaming/flink-streaming-scala/src/main/scala/org/apache/flink/streaming/api/scala/DataStream.scala

```diff
@@ -727,13 +727,118 @@ class DataStream[T](javaStream: JavaStream[T]) {
    * every element of the DataStream the result of .toString
    * is written.
    *
+   * @param writeMode
+   * Control the behavior for existing files. Options are
+   * NO_OVERWRITE and OVERWRITE.
+   *
+   */
+  def writeAsText(
+      path: String,
+      writeMode: FileSystem.WriteMode): DataStreamSink[T] = {
+    if (writeMode != null) {
+      javaStream.writeAsText(path, writeMode)
+    } else {
+      javaStream.writeAsText(path)
+    }
+  }
+
+  /**
+   * Writes a DataStream to the file specified by path in text format. The
+   * writing is performed periodically, in every millis milliseconds. For
+   * every element of the DataStream the result of .toString
+   * is written.
+   *
+   * @param writeMode
+   * Control the behavior for existing files. Options are
+   * NO_OVERWRITE and OVERWRITE.
+   *
+   */
+  def writeAsText(
+      path: String,
+      writeMode: FileSystem.WriteMode,
+      millis: Long): DataStreamSink[T] = {
+    if (writeMode != null) {
+      javaStream.writeAsText(path, writeMode, millis)
+    } else {
+      javaStream.writeAsText(path, millis)
+    }
+  }
+
+  /**
+   * Writes a DataStream to the file specified by path in text format. The
```

Comment:
`writeToCsv` does not write `in text format` and does also not use the `.toString` method.
Please fix the documentation.